lib/cb/models/implementations/saved_search.rb

```diff
@@ -36,18 +36,19 @@ def initialize(args = {})
         @search_parameters          = SearchParameters.new(args['SavedSearchParameters'] || {})
       end
 
-      def create_to_xml
-        <<-eos
-          <Request>
-            <HostSite>#{host_site}</HostSite>
-            <Cobrand>#{cobrand}</Cobrand>
-            <SearchName>#{search_name}</SearchName>
-            #{search_parameters.to_xml}
-            <IsDailyEmail>#{is_daily_email.to_s.upcase}</IsDailyEmail>
-            <ExternalUserID>#{external_user_id}</ExternalUserID>
-            <DeveloperKey>#{Cb.configuration.dev_key}</DeveloperKey>
-          </Request>
-        eos
+      def create_to_json
+        hash = {
+          'SiteID' => site_id,
+          'Cobrand' => cobrand,
+          'EmailDeliveryDay' => email_delivery_day,
+          'IsDailyEmail' => is_daily_email.to_s.upcase,
+          'userOAuthToken' => user_oauth_token,
+          'HostSite' => host_site,
+          'SearchName' => search_name,
+          'SavedSearchParameters' => search_parameters.to_hash
+        }
+        hash['EmailDeliveryDay'] = email_delivery_day unless is_daily_email
```

Comment:
Oh man, I should read better. I didn't notice that it was also set when the hash is created. I feel like since it's set during instantiation the `unless` part here might not have the intended effect. It'll be set once no matter once, then if `is_daily_email` is false/nil it'll be set again to the same value. If `is_daily_email` is true/non-nil, we'll skip the second assignment and it'll still be set upon instantiation (on line 43).